command/meta.go

```diff
@@ -0,0 +1,202 @@
+package command
+
+import (
+	"bufio"
+	"bytes"
+	"flag"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/mitchellh/cli"
+	text "github.com/tonnerre/golang-text"
+)
+
+// maxLineLength is the maximum width of any line.
+const maxLineLength int = 72
+
+// FlagSetFlags is an enum to define what flags are present in the
+// default FlagSet returned.
+type FlagSetFlags uint
+
+const (
+	FlagSetNone FlagSetFlags = iota << 1
+	FlagSetHTTP FlagSetFlags = iota << 1
+)
+
+type Meta struct {
+	Ui    cli.Ui
+	Flags FlagSetFlags
+
+	flagSet *flag.FlagSet
+
+	// These are the options which correspond to the HTTP API options
+	httpAddr   string
+	datacenter string
+	token      string
+	stale      bool
+}
+
+// HTTPClient returns a client with the parsed flags. It panics if the command
+// does not accept HTTP flags or if the flags have not been parsed.
+func (m *Meta) HTTPClient() (*api.Client, error) {
+	if !m.hasHTTP() {
+		panic("no http flags defined")
+	}
+	if !m.flagSet.Parsed() {
+		panic("flags have not been parsed")
+	}
+
+	return api.NewClient(&api.Config{
```

Comment:
I don't think this will respect the environment variables such as for the token. I think we need to start with an `api.DefaultConfig()` and then optionally poke in the command line variables if they are present. See https://github.com/hashicorp/consul/pull/2610.